cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -215,50 +215,58 @@ func (s *Server) newRepo(ctx context.Context, targetName types.NamespacedName, u
 					dc.Server, dc.Username, dc.Password, dc.Email, authStr)
 				secret.Data[".dockerconfigjson"] = []byte(base64.StdEncoding.EncodeToString([]byte(configStr)))
 			} else {
-				return status.Errorf(codes.Internal, "Docker credentials configuration is missing")
+				return nil, status.Errorf(codes.Internal, "Docker credentials configuration is missing")
 			}
 		default:
-			return status.Errorf(codes.Internal, "Unexpected package repository authentication type: %q", auth.Type)
+			return nil, status.Errorf(codes.Internal, "Unexpected package repository authentication type: %q", auth.Type)
 		}
 	}
 
-	secretRef := ""
+	secretRef, checkSecret := "", false
 	if secret != nil {
 		// create a secret first, if applicable
 		if typedClient, err := s.clientGetter.Typed(ctx, s.kubeappsCluster); err != nil {
-			return err
+			return nil, err
 		} else if secret, err = typedClient.CoreV1().Secrets(targetName.Namespace).Create(ctx, secret, metav1.CreateOptions{}); err != nil {
-			return statuserror.FromK8sError("create", "secret", secret.GetName(), err)
+			return nil, statuserror.FromK8sError("create", "secret", secret.GetName(), err)
 		} else {
 			secretRef = secret.GetName()
 		}
 	} else if tlsConfig != nil && tlsConfig.GetSecretRef().GetName() != "" {
 		// check that the specified secret exists
-		secretRef = tlsConfig.GetSecretRef().GetName()
-		if typedClient, err := s.clientGetter.Typed(ctx, s.kubeappsCluster); err != nil {
-			return err
-		} else if _, err = typedClient.CoreV1().Secrets(targetName.Namespace).Get(ctx, secretRef, metav1.GetOptions{}); err != nil {
-			return statuserror.FromK8sError("get", "secret", secretRef, err)
-		}
+		secretRef, checkSecret = tlsConfig.GetSecretRef().GetName(), true
 	} else if auth != nil && auth.GetSecretRef().GetName() != "" {
+		secretRef, checkSecret = auth.GetSecretRef().GetName(), true
+	}
+
+	if secretRef != "" && checkSecret {
 		// check that the specified secret exists
-		secretRef = auth.GetSecretRef().GetName()
 		if typedClient, err := s.clientGetter.Typed(ctx, s.kubeappsCluster); err != nil {
-			return err
+			return nil, err
 		} else if _, err = typedClient.CoreV1().Secrets(targetName.Namespace).Get(ctx, secretRef, metav1.GetOptions{}); err != nil {
-			return statuserror.FromK8sError("get", "secret", secretRef, err)
+			return nil, statuserror.FromK8sError("get", "secret", secretRef, err)
 		}
+		// TODO (gfichtenholt) also check that the secret type corresponds to specified auth type,
+		// e.g. if AuthType is PackageRepositoryAuth_PACKAGE_REPOSITORY_AUTH_TYPE_BASIC_AUTH,
+		// check that the secret has "username" and "password" fields, etc.
```

Comment:
Not sure if it's useful, but kubernetes also has actual secret types that - such as `kubernetes.io/basic-auth` - that cause certain validation to be done on the data (ie. ensuring that the "username" and "password" fields are present). I think we already chatted about these a while back when looking at the design, but worth checking so we don't double-up. https://kubernetes.io/docs/concepts/configuration/secret/#secret-types

Reply:
yes, I know about those. I don't know if flux authors do. Their examples 
https://fluxcd.io/docs/components/source/helmrepositories/#spec-examples
only show opaque secrets. That doesn't necessarily mean they don't support typed secrets, I just need to test it out. It feels  kind of weird for them not to use typed secrets in examples - would have been easier to understand, but I digress. I will add a TODO for this

Reply:
Yep, sounds good. I think the typed secrets are a more recent and unenforced, so many code bases (including ours) haven't updated to use them yet.